golem-api-grpc/proto/golem/apidefinition/api_definition.proto

```diff
@@ -79,4 +80,10 @@ message CompiledWorkerBinding {
   optional golem.rib.RibByteCode compiled_idempotency_key_expr = 9;
   optional golem.rib.RibInputType idempotency_key_rib_input = 10;
   optional golem.rib.WorkerFunctionsInRib worker_functions_in_response = 11;
+  WorkerBindingType type = 12;
```

Comment:
Could we make sure it is backward compatible?

Reply:
I would make it `optional` just to be safe :)

Reply:
Do you care about backwards compat of the grpc api? I was under the impression it was intended only for use between golem components (and I assume you don't support running mixed versions of golem)

Just double checking. If yes, I can absolutely change this (and the component api as well -- I think there is another case like this there)

Reply:
For a variaty of reasons, this proto definition is used to store in repo.

Example:

```rust
    pub fn deserialize(bytes: &[u8]) -> Result<Vec<CompiledRoute>, String> {
        let (version, data) = bytes.split_at(1);

        match version[0] {
            SERIALIZATION_VERSION_V1 => {
                let proto_value: CompiledHttpApiDefinition = Message::decode(data)
                    .map_err(|e| format!("Failed to deserialize value: {e}"))?;

                let value = proto_value
                    .routes
                    .into_iter()
                    .map(CompiledRoute::try_from)
                    .collect::<Result<Vec<CompiledRoute>, String>>()?;

                Ok(value)
            }
            _ => Err("Unsupported serialization version".to_string()),
        }
    }
    
        pub fn serialize(value: &[CompiledRoute]) -> Result<Bytes, String> {
        let routes: Vec<CompiledHttpRoute> = value
            .iter()
            .cloned()
            .map(CompiledHttpRoute::try_from)
            .collect::<Result<Vec<CompiledHttpRoute>, String>>()?;

        let proto_value: CompiledHttpApiDefinition = CompiledHttpApiDefinition { routes };

        let mut bytes = BytesMut::new();
        bytes.put_u8(SERIALIZATION_VERSION_V1);
        bytes.extend_from_slice(&proto_value.encode_to_vec());
        Ok(bytes.freeze())
    }
```

Reply:
I think everything related to grpc for worker gateway (Api Definition etc) has to be made backward compatible